### DISCUSSION TITLE
How to check if object is present in array of objects?

### DISCUSSION THREAD

**Discussion by finlaydotb** [Orange-OpenSource/hurl#1888] [Q&A]
I am using https://hurl.dev/ for testing, and I have an endpoint that returns an array of objects.

Now I want to assert that this array of object contains a particular array. 

How do I do this?

I have tried

```
jsonpath "$" includes '{"id":"123"}'
```

got the error

```
 69 | jsonpath "$" includes '{"id":"123"}'
    |                       ^ invalid predicate value

```

tried

```
jsonpath "$" includes "{\"id\":\"123\"}"
```

Got the error

```
 69 | jsonpath "$" includes "{\"id\":\"123\"}"
    |   actual:   [object]
    |   expected: includes string <{"id":"123"}>
    |

```

This seems to be closer to getting it, but looks like there is a mismatch in types? Also how do I avoid having to manually escape the json string myself.

To try and get the type correct and also not to manually escape the json, I tried

```
jsonpath "$" includes json```{"id":"123"}```
```

but got the error

```
 69 | jsonpath "$" includes json```{"id":"123"}```
    |                       ^ invalid predicate value

```

So question is, if I have an endpoint that returns an array of objects, how do I assert that the response contains/includes a particular object?

**Comment by jcamiel** [comment]
Hi @finlaydotb, sorry for the late answer.

Currently, you can only test for "leaf" in asserts (the right side XXX of this: `jsonpath "$" includes '{"id":"123"}'`) . 

You can change your test from this: `jsonpath "$" includes "{\"id\":\"123\"}"` to this:

`jsonpath "$.id" == "123"`

You can also use [filters](https://hurl.dev/docs/filters) on JSONPath queries:

`jsonpath "$.books" nth 0 == "Dune"`

Or 

`jsonpath "$.books[0]" == "Dune"`

This can be seems as restrictive but for the moment we don't have good syntax/spec for what you want. We need to define what would be  a right answer if the test has been `jsonpath "$" includes "{\"id\":\"123\", \"name\":\"foo\"}"` and the response is actually `{"name":"foo","id":"123"}` for instance.
